src/main/Main.hs

```diff
@@ -283,16 +283,9 @@ commandLineHandler progName isInterpreter = complicatedOptions
                     updateCmd
                     (pure ())
         addCommand' "upgrade"
-                    "Upgrade to the latest stack (experimental)"
+                    "Upgrade to the latest stack"
                     upgradeCmd
-                    ((,) <$> switch
-                              ( long "git"
-                             <> help "Clone from Git instead of downloading from Hackage (more dangerous)" )
-                         <*> strOption
-                              ( long "git-repo"
-                             <> help "Clone from specified git repository"
-                             <> value "https://github.com/commercialhaskell/stack"
-                             <> showDefault ))
+                    (upgradeOpts Nothing) -- FIXME insert the default platform to use
```

Comment:
I think it makes sense to use the same logic as is currently used to figure out the platform.  Since we have a configuration with this command, perhaps `asks getPlatform` is a sufficient default?

This info does indeed correspond to the build platform, but is overidable in the case of cross compilation - https://github.com/commercialhaskell/stack/blob/master/src/Stack/Config.hs#L254 .